experimental/examples/bubblebench/sharedtree/src/proxy/tree.ts

```diff
@@ -129,7 +130,7 @@ export class TreeArrayProxy<T> implements IArrayish<T> {
     toString(): string { return this.items.toString(); }
     toLocaleString(): string { return this.items.toLocaleString(); }
 
-    pop(): T | undefined {
+    pop(): Serializable<T> | undefined {
```

Comment:
does this provide any safety, or should we just return T?

Reply:
You're right that there is no additional safety since the program fails to compile if `T !== Serializable<T>`.

I think the advantage of consistently typing as Serializable<T> in this case is that keeps the API signatures for symmetric in the documentation.

If I saw an API like this:
```ts
push(value: Serializable<T>): void;
pop(): T;
```

It would make me wonder if the class is doing something weird, like implicitly deserializing.  (...which I know makes no sense, but I'd still wonder what I was missing?)